phd-tests/framework/src/disk/crucible.rs

```diff
@@ -62,9 +62,70 @@ impl Drop for Downstairs {
 /// An RAII wrapper around a Crucible disk.
 #[derive(Debug)]
 pub struct CrucibleDisk {
-    /// The name to use in instance specs that include this disk.
     device_name: DeviceName,
+    inner: Mutex<Inner>,
+}
+
+impl CrucibleDisk {
+    #[allow(clippy::too_many_arguments)]
+    pub(crate) fn new(
+        device_name: DeviceName,
+        min_disk_size_gib: u64,
+        block_size: BlockSize,
+        downstairs_binary_path: &impl AsRef<std::ffi::OsStr>,
+        downstairs_ports: &[u16],
+        data_dir_root: &impl AsRef<Path>,
+        read_only_parent: Option<&impl AsRef<Path>>,
+        guest_os: Option<GuestOsKind>,
+        log_mode: ServerLogMode,
+    ) -> anyhow::Result<Self> {
+        Ok(Self {
+            device_name,
+            inner: Mutex::new(Inner::new(
+                min_disk_size_gib,
+                block_size,
+                downstairs_binary_path,
+                downstairs_ports,
+                data_dir_root,
+                read_only_parent,
+                guest_os,
+                log_mode,
+            )?),
+        })
+    }
+
+    /// Obtains the current volume construction request for this disk.
+    pub fn vcr(&self) -> VolumeConstructionRequest {
+        self.inner.lock().unwrap().vcr()
+    }
+
+    /// Sets the generation number to use in subsequent calls to create a
+    /// backend spec for this disk.
+    pub fn set_generation(&self, generation: u64) {
+        self.inner.lock().unwrap().generation = generation;
+    }
+}
+
+impl super::DiskConfig for CrucibleDisk {
+    fn device_name(&self) -> &DeviceName {
+        &self.device_name
+    }
+
+    fn backend_spec(&self) -> ComponentV0 {
+        self.inner.lock().unwrap().backend_spec()
+    }
+
+    fn guest_os(&self) -> Option<GuestOsKind> {
+        self.inner.lock().unwrap().guest_os()
+    }
+
+    fn as_crucible(&self) -> Option<&CrucibleDisk> {
+        Some(self)
+    }
+}
 
+#[derive(Debug)]
+struct Inner {
     /// The UUID to insert into this disk's `VolumeConstructionRequest`s.
     id: Uuid,
```

Comment:
since the UUID will not change across VCRs, perhaps it ought _not_ be inside the mutex, forcing it to be immutable?

not actually a big deal as i don't _think_ we need to access it anywhere that we could avoid taking a lock, but it might express intent a bit better?

Reply:
I've pulled some of this stuff out of `Inner` in 1f04b2f.